workers/unity/Packages/io.improbable.gdk.buildsystem/Util/CompilerValidator.cs

```diff
@@ -0,0 +1,29 @@
+using System.Linq;
+using UnityEditor;
+using UnityEditor.Compilation;
+
+namespace Packages.io.improbable.gdk.buildsystem.Util
+{
+    [InitializeOnLoad]
+    public static class CompilerValidator
+    {
+        private static bool hasCompileErrors;
+        private static int compileErrorCount;
+
+        public static bool HasCompileErrors => hasCompileErrors;
+
+        public static int CompileErrorCount => compileErrorCount;
+
+        static CompilerValidator()
+        {
+            CompilationPipeline.assemblyCompilationFinished += ProcessCompileFinished;
```

Comment:
As this is a callback for a single assembly finishing compilation, could we have ordering problems that cause logic errors here? That is to say: 

1. Assemblies A, B, and C kick off compilation at the same time.
2. Assembly A finishes w/o errors
3. Assembly B finishes w/ errors.
4. Assembly C finishes w/o errors.

I think right now, `HasCompileErrors` would be false for this case. 

You may need to listen for [`compilationStarted`](https://docs.unity3d.com/ScriptReference/Compilation.CompilationPipeline-compilationStarted.html) and [`compilationFinished`](https://docs.unity3d.com/ScriptReference/Compilation.CompilationPipeline-compilationFinished.html) to merge assembly compile results and reset state.

Reply:
That's true. I'll need to rework how I check for the errors in the assemblies. I'll look into those two events 👍